### PR TITLE
KFP Cache Cert Manager patch

### PR DESCRIPTION
**Pull request by akartsky**
This PR fixes the KFP cache and uses cert-manager so that it works on EKS

### CODE REVIEW

**Review by rrrkharse — APPROVED** [review]

**Review by rrrkharse — APPROVED** [review]

**Comment by rrrkharse** [comment]
Need to update option 2 here:
https://github.com/awslabs/kubeflow-manifests/blob/main/website/content/en/docs/deployment/cognito/guide.md#40-building-manifests-and-deploying-kubeflow

**Comment by rrrkharse** [comment]
Same here for option 2:
https://github.com/awslabs/kubeflow-manifests/blob/main/website/content/en/docs/deployment/cognito-rds-s3/guide.md#configure-custom-domain-and-cognito

**Comment by rrrkharse** [comment]
Vanilla KF individual component instructions needs update:
https://github.com/awslabs/kubeflow-manifests/blob/main/website/content/en/docs/deployment/vanilla/guide.md#kubeflow-pipelines

**Comment by akartsky** [comment]
Ya i had already changed those 2 md files 
Maybe refresh the PR 

https://github.com/awslabs/kubeflow-manifests/pull/249/files#diff-a4821e7efaf96d2294ed36cca3adb24c60f4fc07af891663c43ec3afa1154e05

**Comment by rrrkharse** [comment]
Right, actually https://github.com/awslabs/kubeflow-manifests/blob/main/website/content/en/docs/deployment/cognito-rds-s3/guide.md#configure-custom-domain-and-cognito doesn't need changes so ignore my comment on that

**Review by rrrkharse — APPROVED** [review]
LGTM

File: awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-webhook-config.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: admissionregistration.k8s.io/v1
```
Comment (goswamig) [inline awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-webhook-config.yaml:1]: Is this coming from somewhere ?

File: awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-cert.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: cert-manager.io/v1
```
Comment (goswamig) [inline awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-cert.yaml:1]: Is this coming from open source repo ?

File: awsconfigs/common/pipeline/multi-user/delete-cache-deployer.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (goswamig) [reply awsconfigs/common/pipeline/multi-user/delete-cache-deployer.yaml:1]: or we don't need it in cert manager?

**Review by goswamig — APPROVED** [review]
some question take a look.

File: awsconfigs/common/pipeline/multi-user/delete-cache-deployer.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (akartsky) [reply awsconfigs/common/pipeline/multi-user/delete-cache-deployer.yaml:1]: We don't need any of the `deployer` related stuff 
All of this is done by cert-manager

File: awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-webhook-config.yaml (hunk 1 above)
Comment (akartsky) [reply awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-webhook-config.yaml:1]: `deployer` used to create the webhook before 

https://github.com/kubeflow/pipelines/blob/master/backend/src/cache/deployer/cache-webhook-config.v1.yaml.template

File: awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-cert.yaml (hunk 2 above)
Comment (akartsky) [reply awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-cert.yaml:1]: cert-manager is being deployed by the open source repo 

The `Certificate` yaml is constructed using the cert-manager documentation 
https://cert-manager.io/docs/tutorials/acme/dns-validation/

**Review by ryansteakley — APPROVED** [review]

File: awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-cert-issuer.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: kfp-cache-selfsigned-issuer
+spec:
+  selfSigned: {}
```
Comment (ryansteakley) [inline awsconfigs/common/pipeline/multi-user/cache-deployer-cert-manager/cache-cert-issuer.yaml:6]: What does this selfSigned being empty indicate?

**Pull request merged**